File: frontend/src/types/release.ts
import type { Shoe, ShoeDetail } from "./shoe";

export type ReleaseStatus = "UPCOMING" | "ONGOING" | "ENDED";

export interface ReleaseSize {
  id: number;
  size: number;
  stock: number;
  price: number;
}

export interface Release {
  id: number;
  shoe: Pick<Shoe, "id" | "name" | "brand" | "modelNumber"> & {
    imageUrl: string;
  };
  releaseDateTime: string;
  endDateTime: string;
  status: ReleaseStatus;
  totalStock: number;
}

export interface ReleaseDetail {
  id: number;
  shoe: ShoeDetail;
  releaseDateTime: string;
  endDateTime: string;
  status: ReleaseStatus;
  totalStock: number;
  sizes: ReleaseSize[];
}

export interface ReleaseListParams {
  status?: ReleaseStatus;
  page?: number;
  size?: number;
}

File: backend/kickrush-api/src/main/fixture/java/com/kanga/kickrushapi/mock/Release.java
package com.kanga.kickrushapi.mock;

import java.time.LocalDateTime;
import java.util.List;

public record Release(Long id, Long shoeId, LocalDateTime releaseDateTime, LocalDateTime endDateTime,
                      ReleaseStatus status, int totalStock, List<ReleaseSize> sizes) {

    public static Release sample(Long id, Long shoeId, LocalDateTime releaseDateTime, LocalDateTime endDateTime,
                                 ReleaseStatus status, int totalStock, List<ReleaseSize> sizes) {
        return new Release(id, shoeId, releaseDateTime, endDateTime, status, totalStock, sizes);
    }
}
